pkg/hintrunner/zero/zerohint_others.go

```diff
@@ -82,30 +82,48 @@ func createVMExitScopeHinter() (hinter.Hinter, error) {
 	}, nil
 }
 
-// MemcpyEnterScope hint enters a new scope for the memory copy operation with a specified length
+// MemEnterScope hint enters a new scope for the memory copy/set operation with a specified value
 //
-// `newMemcpyEnterScopeHint` takes 1 operander as argument
-//   - `len` is the length value that is added in the new scope
-func newMemcpyEnterScopeHint(len hinter.ResOperander) hinter.Hinter {
+// `newMemEnterScopeHint` takes 2 operanders as arguments
+//   - `value` is the value that is added in the new scope
+//   - `memset` specifies whether it's a memset or memcpy operation
+func newMemEnterScopeHint(value hinter.ResOperander, memset bool) hinter.Hinter {
+	var name string
+	if memset {
+		name = "MemsetEnterScope"
+	} else {
+		name = "MemcpyEnterScope"
+	}
 	return &GenericZeroHinter{
-		Name: "MemcpyEnterScope",
+		Name: name,
 		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
-			//>  vm_enter_scope({'n': ids.len})
-			len, err := hinter.ResolveAsFelt(vm, len)
+			// MemsetEnterScope
+			//> vm_enter_scope({'n': n})
```

Comment:
A detail on this comment: The value assigned here is `ids.n`. For a moment I thought we'd have to change the code 😅 , until I checked with the [source code](https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/memset.cairo#L11)

Reply:
Ah bad comment 🥲